cms/djangoapps/contentstore/tests/tests.py

```diff
@@ -308,37 +159,30 @@ def test_inactive_session_timeout(self):
         resp = self.client.get_html(course_url)
 
         # re-request, and we should get a redirect to login page
-        self.assertRedirects(resp, settings.LOGIN_URL + '?next=/home/')
+        self.assertRedirects(resp, settings.LOGIN_URL + '?next=/home/', target_status_code=302)
 
-    @mock.patch.dict(settings.FEATURES, {"ALLOW_PUBLIC_ACCOUNT_CREATION": False})
-    def test_signup_button_index_page(self):
+    @data(
+        (True, 'assertContains'),
+        (False, 'assertNotContains'))
+    @unpack
+    def test_signin_and_signup_buttons_index_page(self, allow_account_creation, assertion_method_name):
         """
         Navigate to the home page and check the Sign Up button is hidden when ALLOW_PUBLIC_ACCOUNT_CREATION flag
-        is turned off
-        """
-        response = self.client.get(reverse('homepage'))
-        self.assertNotContains(response, '<a class="action action-signup" href="/signup">Sign Up</a>')
-
-    @mock.patch.dict(settings.FEATURES, {"ALLOW_PUBLIC_ACCOUNT_CREATION": False})
-    def test_signup_button_login_page(self):
-        """
-        Navigate to the login page and check the Sign Up button is hidden when ALLOW_PUBLIC_ACCOUNT_CREATION flag
-        is turned off
-        """
-        response = self.client.get(reverse('login'))
-        self.assertNotContains(response, '<a class="action action-signup" href="/signup">Sign Up</a>')
-
-    @mock.patch.dict(settings.FEATURES, {"ALLOW_PUBLIC_ACCOUNT_CREATION": False})
-    def test_signup_link_login_page(self):
-        """
-        Navigate to the login page and check the Sign Up link is hidden when ALLOW_PUBLIC_ACCOUNT_CREATION flag
-        is turned off
+        is turned off, and not when it is turned on.  The Sign In button should always appear.
         """
-        response = self.client.get(reverse('login'))
-        self.assertNotContains(
-            response,
-            '<a href="/signup" class="action action-signin">Don&#39;t have a Studio Account? Sign up!</a>'
-        )
+        with mock.patch.dict(settings.FEATURES, {"ALLOW_PUBLIC_ACCOUNT_CREATION": allow_account_creation}):
```

Comment:
Do we even want to keep this setting around anymore?

Reply:
I think the community may use this.  If we want to remove it, I think it should be in a separate PR to not cause any more trouble with this one. :)